java/ql/lib/semmle/code/java/frameworks/apache/Codec.qll

```diff
@@ -0,0 +1,135 @@
+        "org.apache.commons.codec.binary;StringUtils;true;getBytesIso8859_1;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;getBytesUnchecked;(String,String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;getBytesUsAscii;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;getBytesUtf16;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;getBytesUtf16Be;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;getBytesUtf16Le;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;getBytesUtf8;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;newString;(byte[],String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;newStringIso8859_1;(byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;newStringUsAscii;(byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;newStringUtf16;(byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;newStringUtf16Be;(byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;newStringUtf16Le;(byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;newStringUtf8;(byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;DigestUtils;(MessageDigest);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;getDigest;(String,MessageDigest);;Argument[1];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;getMessageDigest;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,ByteBuffer);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,File);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,InputStream);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,Path,OpenOption[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,RandomAccessFile);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,byte[]);;Argument[0];ReturnValue;taint;",
```

Comment:
All these models work on the assumption that `MessageDigest` carries taint, but I'm not sure that that could happen.

Good thing that all the hashing methods themselves weren't modeled, though, since the taint is lost in a hash.